libs/state/src/lib/signal-state-proxy.ts

```diff
@@ -8,30 +8,33 @@ import {
 import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
 import { Observable } from 'rxjs';
 import { select } from '@rx-angular/state/selections';
+import { RxState } from './rx-state.service';
 
 export type SignalStateProxy<State extends object> = {
   [K in keyof State]: Signal<State[K]>;
 };
 
 export function createSignalStateProxy<State extends object>(
   state$: Observable<State>,
-  stateFn: <K extends keyof State>(k: K) => State[K]
+  stateFn: RxState<State>['get']
```

Comment:
Could you explain the type change here? I'm not sure why stateFn becomes type of RxState.get